packages/react-router-dom/modules/Link.js

```diff
@@ -52,9 +57,24 @@ class Link extends React.Component {
     }
   }
 
-  render() {
-    const { replace, to, ...props } = this.props // eslint-disable-line no-unused-vars
+  absolute() {
+    const { to } = this.props
+    const { match } = this.context.router
+    const base = (match && match.url) ? match.url : ''
+    return resolveLocation(to, base)
+  }
+
+  componentWillMount() {
+    this.unlisten = this.context.router.listen(() => { this.forceUpdate() })
```

Comment:
Oh I see, it is because `context` updates in React can be broken by sCU. Ah well, we'll have to wait until React 17 for that to be addressed…

Alternatively, we could define relative `<Link/>`s to be relative to the current URL and not their parent (contrary to `<Route/>`s). That way, you can leave them as-is and no listeners are necessary. I'm having a hard time coming up with examples where that would be bad.